be/src/env/env_broker.cpp

```diff
@@ -469,29 +462,27 @@ Status EnvBroker::sync_dir(const std::string& dirname) {
     return Status::NotSupported("BrokerEnv::sync_dir");
 }
 
-Status EnvBroker::is_directory(const std::string& path, bool* is_dir) {
+StatusOr<bool> EnvBroker::is_directory(const std::string& path) {
     TBrokerFileStatus stat;
     RETURN_IF_ERROR(_list_file(path, &stat));
-    *is_dir = stat.isDir;
-    return Status::OK();
+    return stat.isDir;
 }
 
 Status EnvBroker::canonicalize(const std::string& path, std::string* file) {
     return Status::NotSupported("BrokerEnv::canonicalize");
 }
 
-Status EnvBroker::_get_file_size(const std::string& path, uint64_t* size) {
+StatusOr<bool> EnvBroker::_get_file_size(const std::string& path) {
     TBrokerFileStatus stat;
     Status st = _list_file(path, &stat);
-    *size = stat.size;
-    return st;
+    return stat.size;
```

Comment:
Size should be set regardless of success or fail, should add another macro definition to cover it;

Reply:
if _list_file failed, old implement will set *size，but not set in new implement